homeassistant/components/image/__init__.py

```diff
@@ -205,8 +205,13 @@ async def get(self, request: web.Request, image_id: str, filename: str):
         )
 
 
-def _generate_thumbnail(original_path, content_type, target_path, target_size):
+def _generate_thumbnail(
+    original_path: pathlib.Path | str,
+    content_type: str,
+    target_path: pathlib.Path | str,
+    target_size: tuple[int, int],
+) -> None:
     """Generate a size."""
     image = ImageOps.exif_transpose(Image.open(original_path))
     image.thumbnail(target_size)
-    image.save(target_path, format=content_type.split("/", 1)[1])
+    image.save(target_path, format=content_type.partition("/")[-1])
```

Comment:
Note that content-type can also include `; blah` tacked on at the end, so recommending another partition to remove the tail after the semicolon, followed by an rstrip.  Other code similar to what I'm asking looks like this: `content_type.partition(";")[0]`.